recipes/h5pp/all/conanfile.py

```diff
@@ -0,0 +1,78 @@
+from conans import ConanFile, CMake, tools
+from conans.tools import download, unzip
+import os, re
+
+class h5ppConan(ConanFile):
+    name = "h5pp"
+    version = "1.8.2"
+    description = "A C++17 wrapper for HDF5 with focus on simplicity"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/DavidAce/h5pp"
+    topics = ("conan","h5pp","hdf5", "binary", "storage")
+    license = "MIT"
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake", "cmake_find_package"
+    requires = "eigen/3.3.7", "spdlog/1.7.0", "hdf5/1.12.0"
+    exports_sources = ["CMakeLists.txt"]
+
+    options = {
+        'tests'     :[True,False],
+        'examples'  :[True,False],
+        'verbose'   :[True,False],
+        }
+
+    default_options = (
+        'tests=False',
+        'examples=False',
+        'verbose=False',
+    )
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        tools.check_min_cppstd(self, "17")
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["H5PP_ENABLE_TESTS"]         = self.options.tests
+            self._cmake.definitions["H5PP_BUILD_EXAMPLES"]       = self.options.examples
+            self._cmake.definitions["H5PP_PRINT_INFO"]           = self.options.verbose
+            self._cmake.definitions["H5PP_DOWNLOAD_METHOD"]      = "conan"
+            self._cmake.configure(source_folder=self._source_subfolder)
+        return self._cmake
+
+    def build(self):
+        if self.options.tests or self.options.examples:
+            cmake = self._configure_cmake()
+            cmake.build()
+
+    def test(self):
+        if self.options.tests:
+            cmake = self._configure_cmake()
+            cmake.build()
+            cmake.test()
+        else:
+            print("Tests have not been enabled. To enable, set option h5pp:tests=True")
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+
+    def package_id(self):
+        self.info.header_only()
```

Comment:
In a `package_info` method, I think you should set the generator names of h5pp.
It's always better to be explicit.
Looking at your test_package, I think it should be:

```python
def package_info(self):
    self.cpp_info.names["cmake_find_package"] = h5pp
    self.cpp_info.names["cmake_find_package_multi"] = h5pp
    self.cpp_info.names["pkg_config"] = h5pp

Reply:
I agree, but I believe that we should be explicit only for generators officially supported by upstream. h5pp doesn't seem to generate a pkg-config file.

Reply:
So I should only remove `self.cpp_info.names["pkg_config"] = h5pp`?. 
Also, shouldn'tit should be ` ... = "h5pp"`?

Reply:
They should be strings indeed, typo.
I added `pkg_config` because I assumed your library was generating one.

Reply:
Thanks! I kept
```python
def package_info(self):
    self.cpp_info.names["cmake_find_package"] = "h5pp"
    self.cpp_info.names["cmake_find_package_multi"] = "h5pp"
```